Feature/edt

nicholsp2 opened the pull request:
## Description

Added a new entry point for "user" to retrieve said user's schedule through a new controller.
Only the courses of the current semester are exposed.

There might be code on the UE entity coming from another branch that was improperly cleaned.

## Checklist

### Implementation

* [x] Files and variables have a name explaining what they do.
* [ ] Updating the DB schemas and seeding the tables do not generates errors.
* [x] All code repeted more than 2 times has been refactored in a dedicated file or function.
* [x] Debugs have been removed : `dd()` | `dump()`

### Tools

* [x] PHP CS Fixer has fixed all files inside the `src` folder.
* [x] All commits have been done using Commitizen.

### Documentation

* [x] All new classes and non-trivial functions have a code documentation.
* [x] There are comments to explain complex parts inside your code.
* [x] If a new folder is created, it has been added and explained inside the "Folder structure" part of the `README.md` file.

nicholsp2 committed:
feat: First part of the exposition of the Asso Event and UE entities

nicholsp2 committed:
feat: Added a new entry point on user to retrieve said user's schedule

ThomasRitaine commented on src/Controller/GetEDTController.php:
```diff
@@ -0,0 +1,36 @@
+    public function __invoke(User $data): User
+    {
+        $repository = $this->getDoctrine()->getRepository(Semester::class);
+        $now = new DateTime('now');
```
Le constructeur appelé sans paramètre crée un objet DateTime avec le timestamp actuel

ThomasRitaine commented on src/Controller/GetEDTController.php:
```diff
@@ -0,0 +1,36 @@
+    {
+        $repository = $this->getDoctrine()->getRepository(Semester::class);
+        $now = new DateTime('now');
+        $currentSemesterCode = $repository->getSemesterOfDate($now)->getCode();
```
Puisque tu ne réutilises pas ta variable `$now`, je te conseille de déclarer ta date directement dans le `$repository->getSemesterOfDate()`

ThomasRitaine commented on src/Controller/GetEDTController.php:
```diff
@@ -0,0 +1,36 @@
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+
+class GetEDTController extends AbstractController
```
Il faudrait ajouter de la documentation de code parce que j'ai du mal à comprendre

ThomasRitaine replied:
Je parlais d'ajouter de la documentation sur la classe. A quoi sert-elle, qui l'appelle ?
Voici un exemple de documentation sur un controller
https://github.com/ungdev/etuutt-api/blob/ad786de1dd5f6bce0d7267fff11b2c9e40f693ba/src/Controller/SoftDeleteController.php#L12-L15

ThomasRitaine commented on src/Entity/User.php:
```diff
@@ -987,6 +1002,17 @@ public function removeUEStarVote(UEStarVote $uEStarVote): self
+            }
+        }
+
+        return $tmp;*/
```
Dead code à enlever

ThomasRitaine commented on src/Entity/User.php:
```diff
@@ -40,6 +41,17 @@
+                'path' => '/user/{id}/edt',
+                'controller' => GetEDTController::class,
+                'normalization_context' => [
+                    'groups' => ['user:read:one:edt'],
```
Que penses-tu de changer le code du groupe en `user-edt:read:one` pour garder la convention `entité+spécification:action:complément` ?

ThomasRitaine reviewed (changes requested):
Super travail Pierre 😉
Est-ce que tu peux ajouter un peu de documentation du code et faire le ménage dans ce que tu mets dans cette PR ? Tu peux toujours re-créer une autre PR avec le restes des modifs que tu veux apporter 👍

nicholsp2 committed:
fix: code modification to comply with pull request remarks

removal of dead code / switched group name from user:read:one:edt to user-edt:read:one / added documentation

ThomasRitaine commented on src/Controller/GetEDTController.php:
```diff
@@ -4,9 +4,8 @@
-use Doctrine\ORM\EntityManagerInterface;
-use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Doctrine\ORM\EntityManagerInterface; //needed to be able to return a class
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController; //needed to be used from platform API configuration (see User.php)
```
Ces composants sont déjà très bien documentés par Symfony, pas besoin d'ajouter d'explication ici 😉

ThomasRitaine commented on src/Repository/SemesterRepository.php:
```diff
@@ -27,8 +26,14 @@ public function __construct(ManagerRegistry $registry)
      * @return Semester Returns the semester in which there is the input date
      */
-    public function getSemesterOfDate(DateTimeInterface $date)
+    public function getSemesterOfDate()
```
La méthode `getSemesterOfDate()` est censée retourner le semestre dans lequel est inclue la date donnée en paramètre, comme indiqué dans la documentation de la méthode. Pourquoi retirer ce paramètre ?

Si tu veux que lorsque cette méthode est appelée sans argument, la date choisie soit celle d'ajourd'hui, alors tu peux le faire avec PHP sans passer par les fonctions PHP `func_num_args()` et `func_get_args($int)`.
Voici comment faire :
`public function getSemesterOfDate(DateTimeInterface $date = new DateTime())`

nicholsp2 committed:
fix(getEdtController-SemesterRepository): removed an unused overload on getSemesterOfDate / added documentation

nicholsp2 committed:
refactor(Asso-Event-Translation-UE): Revert of unwanted files

nicholsp2 committed:
fix(UE): fixes a bug created by a conflic during revert

Commit:
:bug: (User) fixed entity User

/users routes will now return all fields, even if they are empty (in this case there content will be null)

Closes #28
